src/main/java/core/basesyntax/operations/ReturnOperation.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class ReturnOperation implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> storage, FruitTransaction transaction) {
+        storage.put(transaction.getFruit(),
+                storage.getOrDefault(transaction.getFruit(), 0)
+                        + transaction.getQuantity());
+    }

```

Comment:
The `ReturnOperation` class correctly implements the `OperationHandler` interface. However, remember to check the result balance before saving it in the Storage to ensure it remains positive. This applies to all operations that modify the storage, including returns.